packages/transactions/src/signature.ts

```diff
@@ -65,7 +68,7 @@ export function deserializeTransactionAuthField(bytesReader: BytesReader): Trans
     case AuthFieldType.PublicKeyUncompressed:
       return createTransactionAuthField(
         PubKeyEncoding.Uncompressed,
-        deserializePublicKey(bytesReader)
+        uncompressPublicKey(deserializePublicKey(bytesReader).data)
```

Comment:
Is the return value of `deserializePublicKey()` always compressed? Is it safe to call `uncompressPublicKey()` on an already uncompressed public key?

Reply:
In this case yes, since the wire format assumes compressed public-keys (from what I can tell, always reading the same amount of bytes). So this was a bug previously in Stacks.js

Reply:
Shouldn't you be calling `uncompressPublicKey()` under `case AuthFieldType.PublicKeyCompressed` though? This seems backwards

Reply:
This is in the deserializer code. The wire format comes as compressed public-keys, which indicate how they should be used. So the pub-key-encoding here is telling us we need to uncompress.

Reply:
I'm working under this assumption from SIP-005
> The spending field body, which will be the following, depending on the field ID:
0x00 or 0x01: The next 33 bytes are a compressed secp256k1 public key. If the field ID is 0x00, the key will be loaded as a compressed secp256k1 public key. If it is 0x01, then the key will be loaded as an uncompressed secp256k1 public key.

Reply:
Reading the SIP it looks like you are correct. The naming here is just confusing